packages/eas-cli/src/credentials/ios/actions/__tests__/SetupAscApiKey-test.ts

```diff
@@ -0,0 +1,129 @@
+import { asMock } from '../../../../__tests__/utils';
+import { findApplicationTarget } from '../../../../project/ios/target';
+import { confirmAsync, promptAsync } from '../../../../prompts';
+import { getAppstoreMock, testAuthCtx } from '../../../__tests__/fixtures-appstore';
+import { createCtxMock } from '../../../__tests__/fixtures-context';
+import {
+  getNewIosApiMock,
+  testAscApiKey,
+  testAscApiKeyFragment,
+  testCommonIosAppCredentialsFragment,
+  testTargets,
+} from '../../../__tests__/fixtures-ios';
+import { AppStoreApiKeyPurpose } from '../AscApiKeyUtils';
+import { getAppLookupParamsFromContext } from '../BuildCredentialsUtils';
+import { SetupAscApiKey, USE_EXISTING } from '../SetupAscApiKey';
+
+jest.mock('../../../../prompts');
+asMock(confirmAsync).mockImplementation(() => true);
+
+describe(SetupAscApiKey, () => {
+  it('skips setting up a App Store Connect Api Key if it is already configured', async () => {
+    const ctx = createCtxMock({
+      nonInteractive: false,
+      ios: {
+        ...getNewIosApiMock(),
+        getIosAppCredentialsWithCommonFieldsAsync: jest.fn(
+          () => testCommonIosAppCredentialsFragment
+        ),
+      },
+    });
+    const appLookupParams = getAppLookupParamsFromContext(ctx, findApplicationTarget(testTargets));
+    const setupAscApiKeyAction = new SetupAscApiKey(
+      appLookupParams,
+      AppStoreApiKeyPurpose.SUBMISSION_SERVICE
+    );
+    await setupAscApiKeyAction.runAsync(ctx);
+
+    // expect not to create a new App Store Connect Api Key on expo servers
+    expect(asMock(ctx.ios.createAscApiKeyAsync).mock.calls.length).toBe(0);
+    // expect configuration not to be updated with a new App Store Connect Api Key
+    expect(asMock(ctx.ios.updateIosAppCredentialsAsync).mock.calls.length).toBe(0);
+  });
+  it('sets up a App Store Connect Api Key, creating a new one if there are no existing App Store Connect Api Keys', async () => {
```

Comment:
This test name is extraordinarily long, what about shortening it a bit?
```suggestion
  it('sets up an ASC Api Key, creating a new one if there are no existing ASC Api Keys', async () => {
```